docs/idm-concept.md

```diff
@@ -0,0 +1,206 @@
+
+#### Signing artifacts on applications
+
+Applications may want to sign artifacts, such as regular data or ephemeral keys. This will be possible in two different ways:
+
+- Sign with the Session Private Key
+- Sign with the Device Private Key
+
+The first method is less intrusive and happens transparently to the user, but is also less secure. More specifically, a robber who steals an IDM Wallet may still sign artifacts with the Session Private Key as long as a session is valid because that key is stored unencrypted. Verifiers will still see those signatures as valid until the DID owner revokes the stolen device.
+
+The second method is more secure as the user is prompted to unlock IDM to decrypt the Device Private Key, but it is more intrusive. The interaction is similar to the authentication process, where the user is prompted inside the IDM Wallet to sign. Because the Device Private Key is encrypted, this gives the DID owner plenty of time to revoke that key in another IDM Wallet.
+
+Ultimately, the application may choose between both methods for different situations depending on the security degree they want to have.
+
+#### Verifying signatures
+
+Any party might verify the authenticity and authorship of artifacts signed by others.
+
+- The verification of an artifact signed by the Session Public Key is performed by:
+    1. Verifying the signature of the artifact against the Session Public Key of the author.
+    2. Verifying the signature of the Session Public Key against the Device Public Key of the author. This proves that the Session Public Key was authorized by the Device Private Key.
+    3. Verifying if the Device Public Key of the author is present in the DID-Document associated with the DID. This proves that the Device Public Key was authorized by the author's identity.
+    4. If Device Public Key is flagged as revoked, compare its revocation effective date with the signature date
+
+
+- The verification of an artifact signed by the Device Public Key is performed by:
+    1. Verifying the signature of the artifact against the Device Public Key of the author.
+    2. Verifying if the Device Public Key of the author is present in the DID-Document associated with the DID. This proves that the Device Public Key was authorized by the Identity (author).
+    3. If Device Public Key is flagged as revoked, compare its revocation effective date with the signature date
+
+#### Managing application sessions
+
+Users will be able to revoke one or all application sessions from within the identity's applications list. After revoking a session, the application will no longer be able to use that session, effectively stopping it from being able to perform operations that require the Session and Device Private Keys.

```

Comment:
This seems a bit unclear. Are the session keys device dependent? If so, can you manage the application’s sessions in other devices from a given device?

I think it should be “all applications sessions from within the identity's device’s applications list”, because the same applications will have different sessions in different devices for the same identity, correct?. That is, it should be clear if the "identity's applications list" from which the user is revoking a session is relative to a device or not.

Reply:
> because the same applications will have different sessions in different devices for the same identity, correct

Correct, but they will all be synchronized, meaning a device has access to all sessions of an app. This allows a device to revoke not only theirs but others as well. That's why it's possible to remove a app altogether, each will basically remove all the sessions associated to that app.

Reply:
Ok, with that point cleared up I think it makes more sense to me now.
Perhaps we could add to the first sentence, for redundancy and clarity, though I don't feel it's quite necessary now:

"Users will be able to revoke one or all application sessions from within the identity's applications list, regardless of whether the application's revoked session is of the device currently being used by the user or not."

Reply:
Done!